contracts/erc721/UserMintableERC721.sol

```diff
@@ -97,17 +110,36 @@ contract UserMintableERC721 is ERC721Token {
    * by the set up anchorRegistry.
    * @param _merkleRoot bytes32 The root hash of the merkle proof/doc
    * @param _tokenURI string The metadata uri
+   * @param _values string[] The values of the leafs that are being proved
+   * using precise-proofs
+   * @param _salts bytes32[] The salts for the field that is being proved
+   * Will be concatenated for proof verification as outlined in
+   * precise-proofs library.
+   * @param _proofs bytes32[][] Documents proofs that are needed
+   * for proof verification as outlined in precise-proofs library.
    */
   function _mintAnchor(
     address _to,
     uint256 _tokenId,
     uint256 _anchorId,
     bytes32 _merkleRoot,
-    string _tokenURI
-
+    string _tokenURI,
+    string[] _values,
+    bytes32[] _salts,
+    bytes32[][] _proofs
   )
   internal
   {
+    for (uint i = 0; i < mandatoryFields.length; i++) {
+      require(
+        MerkleProofSha256.verifyProof(
+          _proofs[i],
+          _merkleRoot,
+          _hashLeafData(mandatoryFields[i], _values[i], _salts[i])
+        ),
+        mandatoryFields[i]
+      );
+    }
     require(
```

Comment:
I would reorder the `require` statements.
Starting with the lowest cost `require` statement.

In the current setting, a missing anchor registry entry transaction would have to pay the full proof validation before it actually fails.